### PR TITLE
AUTOSWITCH:  Extends weapon auto switch with more options

### PR DESCRIPTION
**Pull request by kai-li-wop**
cg_autoswitch now works as follows:
0 = Never: as before, simply switched off.
1 = Always: switches the weapon every time one is picked up
2 = New: switches to a picked up weapon if it is new and was not in the inventory before
3 = Better: changes to a picked up weapon if it is better than the currently selected weapon
4 = New+Better: switches to a picked up weapon if it is new and was not in the inventory before and if it is better than the currently selected weapon.

The new default value for cg_autoswitch is 4 (New+Better). This is a better value for beginners. Experienced players will want to set cg_autochange to 0 anyway.

The weapon order is determined by the cvar cg_weaponOrder. The current default is "/1/2/4/6/3/7/8/9/5/" and can be customized.

The Autoswitch Weapon menu entry has been removed from the Controls menu and is now located in the Options menu on the Game page.

Thanks to Open Arena for giving the idea and the code to build up on. https://github.com/OpenArena/gamecode/commit/32fa9c763bcbd2b3e84e4beb19b0a1f790e7e318

### CODE REVIEW

**Comment by 3aTmE** [comment]
Hey Kai! As requested i tested the new autoswitch feature. 

## What was tested ?

- all `cg_autoswitch` values. ( They worked like expected )
- Creating a custom order with `cg_weaponOrder` . I completely reversed the Order for testing purposes.  ( Like expected  it stayed at the lowest weapon when picking new weapons up, works also fine! )
- Checked if the values are correct when a autoswitch mode was selected in the settings menu. ( worked like expected )

tl:dr Everything works as expected! Well done! :grin:
